### ISSUE TITLE
Remove the use of jQuery 1 functions deprecated in jQuery 3

### ISSUE THREAD

**Issue by laryn** [backdrop-contrib/colorbox#50]
This module uses the jquery method `.load()` to handle the load event. This method was deprecated, and causes Javascript to crash when used with jQuery 3, which is now the default for Backdrop.

The use of the event handler .`load(function(){})` should be converted to `.on('load', function(){})`

**Comment by bugfolder** [comment]
Library developer's main site: http://www.jacklmoore.com/colorbox/
Library github repo: https://github.com/jackmoore/colorbox
No updates since 2016. So, rather than updating to the latest version of the library, it looks like we should patch the existing library for jQuery 3 compatibility.

**Comment by bugfolder** [comment]
@laryn, I've got a site that uses Colorbox with jQuery3 and my colorbox images are still working. Can you post a code snippet that demonstrates the crash?

**Comment by bugfolder** [comment]
Other deprecated jQuery code found in the module library:

* `.click(`
* `.focus()`

**Comment by argiepiano** [comment]
I see the `click()` method, but the `load()` method used in line 1031 of the version packed with this module is actually the one that's not deprecated. jQuery 1 had two `load()` methods - one to load a page from an href (not deprecated) and one to listen to the load even (deprecated). See https://api.jquery.com/load/

The other instance, the call to load in line 463 is simply a call to the local function load, which is defined in line 912.

So, this is not it, unless `click()` causes the problem, which I doubt (using click in jQuery 3 in general prevented listeners from being created, but doesn't crash JS).

**Comment by laryn** [comment]
Yep, I may have jumped the gun here. I saw (and copied verbatim) @argiepiano's notice about the deprecation because the issue I was seeing seemed suspiciously similar and I perused the .js loosely and saw that `.load(` existed... My issue is different and I retract this with apology. 

You can decide whether to handle any deprecated code that turned up here or not.

Again, sorry about that.

**Comment by bugfolder** [comment]
Seems like the other deprecated calls should still be replaced, though (if possible to maintain B/C). I'll leave this issue open, but change the title.

**Comment by indigoxela** [comment]
Let's start somewhere: https://github.com/backdrop-contrib/colorbox/pull/51

In JS code that ships with this module, we can easily fix problems.

The library code... I see two options for that:

1. Keep an eye on the repo, specifically [this issue (and PR)](https://github.com/jackmoore/colorbox/issues/889) and patch (and use it uncompressed)
2. Add a compat layer js file (like core will also do), to add removed functions back, that the library relies on

**Comment by indigoxela** [comment]
Oh, wait, forgot to mention the third option ... it's described here: https://github.com/backdrop-contrib/jquery_migrate/issues/1

**Comment by indigoxela** [comment]
@bugfolder as you're listed as maintainer ... are you interested in fixing this (step by step)?

**Comment by bugfolder** [comment]
> are you interested in fixing this (step by step)?

Since I'm fairly weak in the JS department, I'd welcome PR (PRs) from someone with more expertise. I see you've submitted a PR to #51 (thanks!); I'll go review that shortly.

**Comment by bugfolder** [comment]
#51 code reviewed, looks good, and I applied the patch to a test site that uses Colorbox, and saw no smoke or flames, so I'll merge that PR. And then reopen this issue to address the library code question.

**Comment by bugfolder** [comment]
> options for that:
> 
> 1. Keep an eye on the repo, specifically [this issue (and PR)](https://github.com/jackmoore/colorbox/issues/889) and patch (and use it uncompressed)
> 2. Add a compat layer js file (like core will also do), to add removed functions back, that the library relies on
> 3. ...https://github.com/backdrop-contrib/jquery_migrate/issues/1

I definitely defer to those with greater expertise in this sort of thing, but if core is adding a compat layer (issue number?), that could be a temporary bridge until the library merges the PR in https://github.com/jackmoore/colorbox/issues/889.

**Comment by indigoxela** [comment]
> ... if core is adding a compat layer (issue number?), that could be a temporary bridge...

Core will need some polyfill as well with its 3rd party libs, but possibly colorbox need more tweaks.

For reference, the core issue: https://github.com/backdrop/backdrop-issues/issues/6404
